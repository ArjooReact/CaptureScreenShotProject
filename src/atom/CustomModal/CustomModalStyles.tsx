import { StyleSheet } from "react-native";
import Dimension from "../../utility/Dimension";
const styles=StyleSheet.create({
    modalStyle:{
        flex:0.5,
        borderRadius:Dimension.ImageBorderRadius
    }
})

export default styles