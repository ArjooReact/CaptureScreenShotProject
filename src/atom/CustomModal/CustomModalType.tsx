export interface CustomModalType{
    isModalVisible:boolean
    children:any
}