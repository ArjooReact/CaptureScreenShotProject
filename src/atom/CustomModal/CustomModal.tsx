import React from 'react'
import {StyleSheet,Modal} from 'react-native'
import { CustomModalType } from './CustomModalType'
import styles from './CustomModalStyles'
const CustomModal:React.FC<CustomModalType>=({isModalVisible,children})=>{
    return(<Modal
        animationType='slide'
        style={styles.modalStyle}
        visible={isModalVisible}
        >
     {children}
    </Modal>)
}
export default CustomModal