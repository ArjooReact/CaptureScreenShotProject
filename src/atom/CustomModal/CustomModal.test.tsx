import renderer from 'react-test-renderer';
import {test, jest, describe} from '@jest/globals';
import {expect, it, beforeEach} from '@jest/globals';
import CustomModal from './CustomModal';
import { View } from 'react-native';
describe('all mock', () => {
  let mockButtonClick = jest.fn();
  let children=()=>{
    return<View></View>
  }
  test('testcase for matching customModal Snapshot==>', () => {
    const AppRef = renderer.create(
      <CustomModal isModalVisible={true} children={children} />,
    );
    expect(AppRef.toJSON()).toMatchSnapshot();
  });

  test('testcase for checking presence of dom element for CustomModal==>', () => {
    const AppRef = renderer.create(
        <CustomModal isModalVisible={true} children={children} />,
    );
    console.log('arzoo test======', AppRef.toJSON());
    let props: any = AppRef.toJSON();
    console.log('Arzoo data...',props)
    expect(AppRef.toJSON()).toBeTruthy();
  });

  test('testcase for checking child element for CustomButton==>', () => {
    const styles = {};
    const AppRef = renderer.create(
        <CustomModal isModalVisible={true} children={children} />,
    );
    let props: any = AppRef.toJSON();
    expect(props?.type).toBe('Modal');
  });
});
