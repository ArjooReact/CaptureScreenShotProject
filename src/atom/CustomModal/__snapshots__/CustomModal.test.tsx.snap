// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`all mock testcase for matching customModal Snapshot==> 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  style={
    {
      "borderRadius": 12,
      "flex": 0.5,
    }
  }
  visible={true}
/>
`;
